### PR TITLE
Add some file to match our github template

### PR DESCRIPTION
**Pull request by Zav**
Added some files and updated the CI / Release yml.
The pipelines will have to be updated.

### CODE REVIEW

File: src/GSoft.Extensions.Configuration.Substitution/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+GSoft.Extensions.Configuration.Substitution.RecursiveConfigurationKeyException
```
Comment (Gcaya) [inline src/GSoft.Extensions.Configuration.Substitution/PublicAPI.Unshipped.txt:1]: Shouldn't all of this go into Shipped.txt?

**Review by Gcaya — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+      - uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: "6.0.x"
+
+      - run: ./Build.ps1
+        shell: pwsh
+        env:
+          NUGET_SOURCE: ${{ secrets.NUGET_SOURCE }}
+          NUGET_API_KEY: ${{ secrets.NUGET_API_KEY }}
```
Comment (asimmon) [inline .github/workflows/publish.yml:25]: Here's the new nuget API key secret for [GSoft-inc organization](https://www.nuget.org/profiles/gsoft-inc):
```suggestion
          NUGET_API_KEY: ${{ secrets.GSOFTINC_NUGET_API_KEY }}
```

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Security Policy
+
+If you'd like to report a vulnerability, please open a [GitHub issue](https://github.com/gsoft-inc/dotnet-library-template/issues).
```
Comment (asimmon) [inline SECURITY.md:3]:
```suggestion
If you'd like to report a vulnerability, please open a [GitHub issue](https://github.com/gsoft-inc/gsoft-extensions-configuration-substitution/issues).
```

File: src/GSoft.Extensions.Configuration.Substitution/GSoft - Backup.Extensions.Configuration.Substitution.csproj [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (asimmon) [inline src/GSoft.Extensions.Configuration.Substitution/GSoft - Backup.Extensions.Configuration.Substitution.csproj:1]: This file should not be commited?

File: src/Directory.Build.props [hunk 5]
```diff
@@ -1,13 +1,13 @@
 <Project>
   <PropertyGroup>
     <Copyright>Copyright © Groupe GSoft inc. $([System.DateTime]::UtcNow.ToString(yyyy))</Copyright>
```
Comment (asimmon) [inline src/Directory.Build.props:3]:
```suggestion
    <Copyright>Copyright © GSoft Group Inc. $([System.DateTime]::UtcNow.ToString(yyyy))</Copyright>
```

File: src/Directory.Build.props [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
-    <Authors>Anthony Simmon</Authors>
+    <Authors>GSoft Group Inc.</Authors>
     <Owners>Groupe GSoft inc.</Owners>
```
Comment (asimmon) [inline src/Directory.Build.props:5]: GSoft Group Inc.

**Review by asimmon — APPROVED** [review]

File: src/Directory.Build.props (hunk 6 above)
Comment (asimmon) [inline src/Directory.Build.props:5]:
```suggestion
    <Owners>GSoft Group Inc.</Owners>
```

**Review by Gcaya — APPROVED** [review]

**Pull request merged**